src/scripts/main.js

```diff
@@ -1,3 +1,100 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+
+const container = document.querySelector('.container');
+
+const appendRowButton = document.querySelector('.append-row');
+
+const removeRowButton = document.querySelector('.remove-row');
+
+const appendColumnButton = document.querySelector('.append-column');
+
+const removeColumnButton = document.querySelector('.remove-column');
+
+const rows = tbody.getElementsByTagName('tr');
+const columns = rows[0].getElementsByTagName('td');
+
+function addRow() {
+  const row = tbody.querySelector('tr').cloneNode(true);
+
+  tbody.append(row);
+}
+
+function removeRow() {
+  tbody.lastElementChild.remove();
+}
+
+function addColumn() {
+  for (const tr of rows) {
+    const td = document.createElement('td');
+
+    tr.append(td);
+  }
+}
+
+function removeColumn() {
+  for (const tr of rows) {
+    tr.lastElementChild.remove();
+  }
+}
+
+container.addEventListener('click', (e) => {
+  if (!e.target.classList.contains('button')) {
+    return;
+  }
+
+  switch (e.target) {
+    case appendRowButton:
+      addRow();
+      break;
+
+    case removeRowButton:
+      removeRow();
+      break;
+
+    case appendColumnButton:
+      addColumn();
+      break;
+
+    case removeColumnButton:
+      removeColumn();
+      break;
+  }
+
+  if (rows.length >= 10) {
+    appendRowButton.setAttribute('disabled', 'disabled');
+
+    return;
+  }
```

Comment:
```suggestion
    appendRowButton.disabled = rows.length >= 10
```
Это все что надо, то же самое и для всех остальных

Reply:
Вынес управление disabled в отдельную функцию, но смущает то, что каждый раз сначала нужно устанавливать его в false